chia/wallet/puzzles/dao_finished_state.clsp

```diff
@@ -26,9 +28,14 @@
       (my_inner_puzhash (recreate_self SINGLETON_STRUCT DAO_FINISHED_STATE_MOD_HASH))
     )
     (list
+      ; [REVIEW - optimization] What's the point of this assertion.  This coin creates its own inner puzhash, it's not like it could lie, right?
+      ; [^] Perhaps the purpose is to make sure that this puzzle can't exist outside of a singleton, but anything that is asserting an announcement
+      ; [^] from this puzzle is also going to have to calculate that it is in fact inside of a singleton.
       (list ASSERT_MY_PUZZLEHASH (wrap_in_singleton SINGLETON_STRUCT my_inner_puzhash))
+      ; [REVIEW - optimization] This is never going to be able to pay out right?  Why not just force this to 0 and save a solution argument?
       (list ASSERT_MY_AMOUNT my_amount)
       (list CREATE_COIN my_inner_puzhash my_amount)
+      ; [REVIEW - optimization] This is the only announcement that this puzzle could create.  Why the special 'f'?  It should probably just be ().
```

Comment:
because we assert that the singleton is making an `f` announcement elsewhere

Reply:
But that thing could assert `()` as well right? What's special about `f`?